libfrpint-1.90.5 build fails

**Describe the bug**
Trying to install libfprint from unstable cause the nixos build to fail.

**To Reproduce**
Steps to reproduce the behavior:
1. Add unstable channel 
2. Install libfprint from the unstable channel

**Additional Context**
I added the log as an attachment but I think the problem is here:
```
In function 'rtp_insert',
    inlined from 'bz_match' at ../libfprint/nbis/bozorth3/bozorth3.c:558:3:
../libfprint/nbis/bozorth3/bozorth3.c:331:8: error: '__builtin_memmove' offset 
[-34359738360, -8] is out of the bounds [0, 160000] of object 'rtp' 

[log-nix.txt](https://github.com/NixOS/nixpkgs/files/5783935/log-nix.txt)
with type 'int *[20000]' [-Werror=array-bounds]
  331 |  *r1-- = *r2--;
      |  ~~~~~~^~~~~~~
../libfprint/nbis/bozorth3/bozorth3.c: In function 'bz_match':
../libfprint/nbis/bozorth3/bozorth3.c:375:14: note: 'rtp' declared here
  375 | static int * rtp[ ROT_SIZE_1 ];
      |              ^~~
```

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.86, NixOS, 20.09.2483.c5c6009fb43 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20201201_5a6ddb3`
 - channels(davide): `""`
 - channels(root): `"nixos-20.09.2483.c5c6009fb43, home-manager-20.09, nixos-unstable-21.03pre261422.1a57d96edd1"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

--- a/pkgs/development/libraries/libfprint/default.nix
+++ b/pkgs/development/libraries/libfprint/default.nix
@@ -44,6 +44,8 @@ stdenv.mkDerivation rec {
     nss
   ];
 
+  NIX_CFLAGS_COMPILE = "-Wno-error=array-bounds";
+
   mesonFlags = [
     "-Dudev_rules_dir=${placeholder "out"}/lib/udev/rules.d"
   ];

